Porting to PyBind

RiccardoRossi opened the pull request:
this is an enormous branch, which we will surely need to close and reopen.

the essential feature is that it ports the kratos core to PyBind11. Most importantly the porting relies on boost::shared_ptr to be gone in favor of Kratos::shared_ptr

loumalouomega commented:
Many conflicts

RiccardoRossi commented:
as i am telling this branch is not to be merged directly. However it gives a guide of how the porting should be done. the conflicts come from the boost::shared_ptr-->Kratos::shared_ptr change

RiccardoRossi committed:
FluidDynamicsApplication

RiccardoRossi committed:
minor modifications to bring up to date the StructuralMechanicsApplication

RiccardoRossi committed:
corrections

RiccardoRossi committed:
beinsging to pybind11

RiccardoRossi committed:
switching from boost to std pointer

RiccardoRossi committed:
moving from boost to std pointers

RiccardoRossi committed:
removing a watch

RiccardoRossi committed:
pybind11 porting

RiccardoRossi committed:
obtaining clean version

philbucher committed:
fixing tests

RiccardoRossi committed:
merging master

RiccardoRossi committed:
merging

RiccardoRossi committed:
fixing logger interface

RiccardoRossi committed:
adding missing BDF scheme

RiccardoRossi committed:
using a list instead of a set

RiccardoRossi committed:
finished porting the application to pybind11

RiccardoRossi committed:
ensuring that the file is correctly closed

RiccardoRossi committed:
ensuring that the file is correctly closed

RiccardoRossi committed:
this correction is needed since a node iterator cannot be converted to a Node::Pointer. We could fix by changing the iterators. to be discussed

RiccardoRossi committed:
merging master

RiccardoRossi committed:
using ptr_iterators instead of iterators. we should decide if this is a good idea

RiccardoRossi committed:
reverting to original version

loumalouomega commented:
People, @RiccardoRossi has prepared the following tutorial to migrate to Pybind https://github.com/KratosMultiphysics/Kratos/wiki/Porting-to-PyBind11---common-steps

RiccardoRossi commented:
please note that there is a branch pushed with that name,
so whatever you do for pybind do it in that branch (take a look at the open
PRs)

On Fri, Mar 9, 2018 at 4:08 PM, Vicente Mataix Ferrándiz <
notifications@github.com> wrote:

> People, @RiccardoRossi <https://github.com/riccardorossi> has prepared
> the following tutorial to migrate to Pybind https://github.com/
> KratosMultiphysics/Kratos/wiki/Porting-to-PyBind11---common-steps
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/KratosMultiphysics/Kratos/pull/1513#issuecomment-371838995>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHr7EV7wuCo08meYVlFfYvCXWEzpXuNqks5tcprcgaJpZM4SG06H>
> .
>



-- 


*Riccardo Rossi*

PhD, Civil Engineer


member of the Kratos Team: www.cimne.com/kratos

Tenure Track Lecturer at Universitat Politècnica de Catalunya,
BarcelonaTech (UPC)

Full Research Professor at International Center for Numerical Methods in
Engineering (CIMNE)


C/ Gran Capità, s/n, Campus Nord UPC, Ed. B0, Despatx 102

(please deliver post to the porters of building C1)

08034 – Barcelona – Spain – www.cimne.com  -

T.(+34) 93 401 56 96 skype: *rougered4*



<http://www.cimne.com/>

<https://www.facebook.com/cimne> <http://blog.cimne.com/>
<http://vimeo.com/cimne> <http://www.youtube.com/user/CIMNEvideos>
<http://www.linkedin.com/company/cimne> <https://twitter.com/cimne>

Les dades personals contingudes en aquest missatge són tractades amb la
finalitat de mantenir el contacte professional entre CIMNE i voste. Podra
exercir els drets d'accés, rectificació, cancel·lació i oposició,
dirigint-se a cimne@cimne.upc.edu. La utilització de la seva adreça de
correu electronic per part de CIMNE queda subjecte a les disposicions de la
Llei 34/2002, de Serveis de la Societat de la Informació i el Comerç
Electronic.

 Imprimiu aquest missatge, només si és estrictament necessari.
<http://www.cimne.com/>

RiccardoRossi committed:
merging master into pybind branch

philbucher commented:
so what is the name of the branch we should push to? I didn't find it

RiccardoRossi committed:
porting to pybind11

RiccardoRossi committed:
porting trilinos app

RiccardoRossi committed:
porting trilinos applicatino

RiccardoRossi committed:
trying to make epetra interface to work

RiccardoRossi committed:
adding a minimal test for resize

RiccardoRossi committed:
adding a wrapper around trilinos matrix and pointers

RiccardoRossi committed:
minimal change to ReadMatrixMarket

RiccardoRossi committed:
silencing a warning

RiccardoRossi committed:
adding test

RiccardoRossi committed:
porting to pybind the ALE application

RiccardoRossi committed:
porting to pybind the convection diffusion application

loumalouomega committed:
Adding using namespace pybind11 to add_geometries

loumalouomega committed:
Missing include

loumalouomega committed:
Exception cannot operate with std::vector

@roigcarlo, you solved this for the KRATOS_WATCH?

loumalouomega committed:
Updating MeshingApplication CMakeLists

loumalouomega committed:
Update main python MeshingApplication

loumalouomega committed:
Update add_processes MeshingApplication (WIP)

loumalouomega committed:
Update add_custom_io MeshingApplication (WIP)

loumalouomega committed:
Update add_custom_strategies MeshingApplication (WIP)

loumalouomega committed:
Update add_custom_utilities MeshingApplication (WIP)

loumalouomega committed:
Update add_meshers MeshingApplication (WIP)

loumalouomega committed:
Correcting custom_strategies

loumalouomega committed:
Pointer for mmg_process

loumalouomega committed:
Correcting processes python export

loumalouomega committed:
Correcting mehsers python export

loumalouomega committed:
Correcting utilities python export

philbucher commented:
@RiccardoRossi it does not compile on my machine (any other branch compiles so the problem is not on my side I think)
Neither in MPI:
~~~
In file included from /home/philippb/software/Kratos_dev/external_libraries/mpi_python/python_interface.cpp:2:0:
/home/philippb/software/Kratos_dev/external_libraries/mpi_python/mpi_python.h:8:31: fatal error: pybind11/pybind11.h: No such file or directory
compilation terminated.
external_libraries/mpi_python/CMakeFiles/mpipython.dir/build.make:62: recipe for target 'external_libraries/mpi_python/CMakeFiles/mpipython.dir/python_interface.cpp.o' failed
make[2]: *** [external_libraries/mpi_python/CMakeFiles/mpipython.dir/python_interface.cpp.o] Error 1
CMakeFiles/Makefile2:1207: recipe for target 'external_libraries/mpi_python/CMakeFiles/mpipython.dir/all' failed
~~~
Nor without MPI:
~~~
In file included from /home/philippb/software/Kratos_dev/kratos/utilities/math_utils.h:26:0,
                 from /home/philippb/software/Kratos_dev/kratos/includes/mortar_classes.h:21,
                 from /home/philippb/software/Kratos_dev/kratos/utilities/exact_mortar_segmentation_utility.h:24,
                 from /home/philippb/software/Kratos_dev/kratos/utilities/exact_mortar_segmentation_utility.cpp:19:
/home/philippb/software/Kratos_dev/kratos/includes/ublas_interface.h:34:42: fatal error: boost/numeric/ublas/vector.hpp: No such file or directory
compilation terminated.
kratos/CMakeFiles/KratosCore.dir/build.make:62: recipe for target 'kratos/CMakeFiles/KratosCore.dir/utilities/exact_mortar_segmentation_utility.cpp.o' failed
~~~

furthermore I get the following cmake warning related to pybind:
~~~
CMake Warning (dev) at kratos/CMakeLists.txt:129 (target_link_libraries):
  Policy CMP0023 is not set: Plain and keyword target_link_libraries
  signatures cannot be mixed.  Run "cmake --help-policy CMP0023" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  The keyword signature for target_link_libraries has already been used with
  the target "Kratos".  All uses of target_link_libraries with a target
  should be either all-keyword or all-plain.

  The uses of the keyword signature are here:

   * cmake_modules/pybind11Tools.cmake:107 (target_link_libraries)

This warning is for project developers.  Use -Wno-dev to suppress it.
~~~

Update: Do I have to change sth in the configure file?

RiccardoRossi commented:
Hi @philbucher you need to

1. download the pybind to a directory
2. add ${CMAKE_CXX_FLAGS} -I/home/rrossi/scratch/pybind11/pybind11/include  " in the configuration file (obviosuly with the corrected paths)

the idea is that we will then distribute pybind11 together with the Kratos, but i haven't done it yet

loumalouomega commented:
In my case I used 

-DPYBIND11_INCLUDE_DIR="/home/vicente/src/pybind11/include/"                                    \

In any case my tip is to first compile the core and activate one by one each application, for example, I had some core things not compiling yesterday

RiccardoRossi commented:
as o now, apart for the core, the applications that should be compiling are

StructuralMechanics
FluidDynamics
ExternalSolvers
ALE
Meshing
convection_diffusion
metis
trilinos
mpi

DO NOT generate the runkratos wrapper. It is currently not ported

philbucher commented:
thx guys!

I think it would make sense to add this info to the pybind wiki

loumalouomega commented:
Yep, I don't know if include the compilation too, is trivial (just cmake.. make)

RiccardoRossi commented:
we were discussing with @roigcarlo to push the pybind to the master as a separate PR.

this will allow to automatically add the include path. This will be a cleaner solution.


on a side note, i just merged the master into the branch. There is one test failing, but the error is a correct one. Could u take a look:

======================================================================
ERROR: test_input (test_materials_input.TestMaterialsInput)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rrossi/kratos_pybind11/Kratos/kratos/tests/test_materials_input.py", line 50, in test_input
    read_materials_process.Factory(test_settings,Model)
  File "/home/rrossi/kratos_pybind11/Kratos/kratos/python_scripts/read_materials_process.py", line 7, in Factory
    return ReadMaterialsProcess(Model, settings["Parameters"])
  File "/home/rrossi/kratos_pybind11/Kratos/kratos/python_scripts/read_materials_process.py", line 46, in __init__
    self._AssignPropertyBlock(materials["properties"][i])
  File "/home/rrossi/kratos_pybind11/Kratos/kratos/python_scripts/read_materials_process.py", line 156, in _AssignPropertyBlock
    prop.SetValue(KratosMultiphysics.CONSTITUTIVE_LAW, constitutive_law.Clone())
RuntimeError: Error: Called the virtual function for Clone

in kratos/sources/constitutive_law.cpp:77:virtual ConstitutiveLaw::Pointer ConstitutiveLaw::Clone() const


Stdout:
Importing    KratosStructuralMechanicsApplication

----------------------------------------------------------------------
Ran 116 tests in 28.415s

FAILED (errors=1, skipped=2, expected failures=2)

loumalouomega commented:
Just one question, it is me or the time to compile with pybind is significantly greater than with boost?

RiccardoRossi commented:
@loumalouomega i would say that for pybind11 the time is significantly smaller. I hope i did not get this wrong. What possibly gets longer is linking time
can u time it?

loumalouomega commented:
It is just my sensation, at least with the first compilation I feel it greater

loumalouomega committed:
Updating CMakeLists.txt in ContactStructuralMechanicsApplication

loumalouomega committed:
Updating main python file in ContactStructuralApplication

philbucher commented:
@RiccardoRossi I will fix the error today

loumalouomega committed:
Updating add_mappers python file in ContactStructuralApplication

loumalouomega committed:
Updating add_custom python file in ContactStructuralApplication

loumalouomega committed:
Updating add_strategies python file in ContactStructuralApplication

loumalouomega committed:
Updating add_processes python file in ContactStructuralApplication

loumalouomega committed:
Updating add_linear_solver python file in ContactStructuralApplication

loumalouomega commented:
ContactStruturalMechanicsApplication ported

RiccardoRossi commented:
@philbucher after the PR there are also other tests failing (the same as in the master). I am refraining myself from pushing my local changes until the master is also working. errors are here:

======================================================================
ERROR: test_TL_2D_quadrilateral (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 459, in test_TL_2D_quadrilateral
    self._solve(mp)
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_TL_2D_triangle (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 419, in test_TL_2D_triangle
    self._solve(mp)
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_TL_3D_hexa (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 510, in test_TL_3D_hexa
    self._solve(mp)
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_UL_2D_quadrilateral (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 588, in test_UL_2D_quadrilateral
    self._solve(mp)
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_UL_2D_triangle (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 547, in test_UL_2D_triangle
    self._solve(mp)
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_UL_3D_hexa (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 640, in test_UL_3D_hexa
    self._solve(mp)
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 128, in _solve
    strategy.Solve()
IndexError: bad index

======================================================================
ERROR: test_compare_TL_UL_2D_triangle (test_patch_test_large_strain.TestPatchTestLargeStrain)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 366, in test_compare_TL_UL_2D_triangle
    self._solve_with_strategy(tl_strategy, tl_lhs, iter)
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_large_strain.py", line 161, in _solve_with_strategy
    strategy.SolveSolutionStep()
IndexError: bad index

======================================================================
ERROR: test_formfinding (test_patch_test_formfinding.TestPatchTestFormfinding)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_formfinding.py", line 128, in test_formfinding
    False) # Do PostProcessing for GiD?
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_formfinding.py", line 115, in _execute_formfinding_test
    self._solve(mp)
  File "/home/rrossi/kratos_pybind11/Kratos/applications/StructuralMechanicsApplication/tests/test_patch_test_formfinding.py", line 91, in _solve
    strategy.Solve()
IndexError: bad index

----------------------------------------------------------------------
Ran 65 tests in 4.668s

philbucher commented:
@RiccardoRossi these one are fixed in #1666

RiccardoRossi committed:
merging master into main. Some tests are failing in the master and in this branch

philbucher commented:
I still cannot compile it, it doesn't find boost for some reason
Could one of you please post your configure file?
Thx

philbucher commented:
I found the problem:
Specifying BOOST_HOME does not work any more, now one has to specify Boost_INCLUDE_DIRS and Boost_LIBRARY_DIRS

philbucher commented:
@RiccardoRossi also this appeared, not sure it it is supposed to be there:
`Linking CXX shared module Kratos.cpython-35m-x86_64-linux-gnu.so`

loumalouomega commented:
Did you included the line I write before?

Did you download and compiled the pybind (it must be compiled)?

El 14 mar. 2018 7:03 p. m., "Philipp Bucher" <notifications@github.com>
escribió:

> I found the problem:
> Specifying BOOST_HOME does not work any more, now one has to specify
> Boost_INCLUDE_DIRS and Boost_LIBRARY_DIRS
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/KratosMultiphysics/Kratos/pull/1513#issuecomment-373119462>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATEMgOh-rfEYrJOVAOpYiAYjHjgQHVQrks5teVtcgaJpZM4SG06H>
> .
>

philbucher commented:
I included the line you (Vicente) posted
Pybind is header only ...

Kratos compiles now, just bringing attention to this

philbucher committed:
Porting MappingApplication

philbucher committed:
removing unnecesary include for MappingApp

loumalouomega commented:
Well, it is header only but it is quite stubborn, when you compile it puts
the files in the right place, that's why I compiled...

El 14 mar. 2018 7:09 p. m., "Philipp Bucher" <notifications@github.com>
escribió:

> I included the line you (Vicente) posted
> Pybind is header only ...
>
> Kratos compiles now, just bringing attention to this
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/KratosMultiphysics/Kratos/pull/1513#issuecomment-373121623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATEMgBG1FFCrRVH6AwwG9tmup7FCPZ57ks5teVzogaJpZM4SG06H>
> .
>

philbucher committed:
fixing Metis and Trilinos (did this ever compile @RiccardoRossi?)

philbucher commented:
hm weird
I just downloaded it and gave the path, then it worked
I guess this will be solved if we ship Kratos tgth with pybind @RiccardoRossi ?

Also @RiccardoRossi I fixed the compileation of Trilinos and Metis, did this compile for you?

RiccardoRossi commented:
@philbucher, @loumalouomega in principle pybind should work as header only. We will distribute it as such once it goes together with kratos. @roigcarlo is on it. 

trilinos and metis were compiling. However i compiled with cotire, so errors are probably coming from this.

my hope is that changes are not painful for "normal" applications.

loumalouomega commented:
In my case it didn't work until I compiled with test, weird, but that
solved my problem.

El 14 mar. 2018 9:01 p. m., "Riccardo Rossi" <notifications@github.com>
escribió:

> @philbucher <https://github.com/philbucher>, @loumalouomega
> <https://github.com/loumalouomega> in principle pybind should work as
> header only. We will distribute it as such once it goes together with
> kratos. @roigcarlo <https://github.com/roigcarlo> is on it.
>
> trilinos and metis were compiling. However i compiled with cotire, so
> errors are probably coming from this.
>
> my hope is that changes are not painful for "normal" applications.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/KratosMultiphysics/Kratos/pull/1513#issuecomment-373155678>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATEMgFx8u-_wm0EIZUOZFUP4fueUr3jwks5teXcggaJpZM4SG06H>
> .
>

roigcarlo committed:
Updating python interface for DEM

roigcarlo committed:
Adding new pybind11 to cmake

roigcarlo committed:
Removing apparitions of boost::python in application

roigcarlo committed:
Removing boost::python from custom_utilities

This is a big mindless change (automatic translation) inside a big change an probably now there
is a much more clean way to deal with lists.

roigcarlo committed:
Merge branch 'switching_to_std_pointer' of https://github.com/KratosMultiphysics/Kratos into switching_to_std_pointer

RiccardoRossi committed:
fixing compilation of ContactStucturalMechanics with cotire

RiccardoRossi committed:
fixing cotire compilation

RiccardoRossi committed:
merging the patch

RiccardoRossi committed:
corrections after merging master

RiccardoRossi committed:
merging master

RiccardoRossi committed:
solving an important flaw in the connectivity_preserve_modeler and in the model_part AddElements AddConditions - should be cherrypicked to the core

RiccardoRossi committed:
merging master into branch and solving some small problems in the ContactStructuralMechanicsApplication

RiccardoRossi committed:
removing non default handler to make it to work

philbucher commented:
@RiccardoRossi I get some cmake warnings:
~~~
CMake Warning (dev) at kratos/CMakeLists.txt:129 (target_link_libraries):
  Policy CMP0023 is not set: Plain and keyword target_link_libraries
  signatures cannot be mixed.  Run "cmake --help-policy CMP0023" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  The keyword signature for target_link_libraries has already been used with
  the target "Kratos".  All uses of target_link_libraries with a target
  should be either all-keyword or all-plain.

  The uses of the keyword signature are here:

   * cmake_modules/pybind11Tools.cmake:107 (target_link_libraries)

This warning is for project developers.  Use -Wno-dev to suppress it.
~~~

Doesn'T only happen with the core, also happens with the applications (ALE and FluidDynamics)

RiccardoRossi commented:
@philbucher to correct this warning we should change e.g.

       target_link_libraries(KratosFluidDynamicsApplication KratosCore )

to

      target_link_libraries(KratosFluidDynamicsApplication PRIVATE KratosCore )

RiccardoRossi commented:
@roigcarlo i think we should do something with the python detection for AppVeyor and TravisCL to compile.

could you take a look?

philbucher committed:
porting EmpireApplication

philbucher committed:
Removing Compiler warnings in ALE and FluidDynamics

philbucher commented:
@RiccardoRossi I did it for FluidDynamics and ALE, but I don't feel confident to to it for the Core, can you please have a look? (Should be the same small change but I am not 100% sure)

RiccardoRossi committed:
removing a cmake warning in the core

RiccardoRossi commented:
@philbucher i pushed the change to the core. Tests run so i hope it is ok...

josep-m-carbonell committed:
Merge branch 'master' into solid/develop

rubenzorrilla committed:
FSI application porting to pybind

rubenzorrilla committed:
Private KratosCore

rubenzorrilla committed:
Merge branch 'feature-FSI-porting-to-pybind' into switching_to_std_pointer

rubenzorrilla committed:
Merge branch 'switching_to_std_pointer' of https://github.com/KratosMultiphysics/Kratos into switching_to_std_pointer

roigcarlo committed:
Using PYBIND11_PYTHON_VERSION to select python version

jcotela commented on CMakeLists.txt:
```diff
@@ -206,14 +206,6 @@ set(Boost_USE_STATIC_LIBS   OFF)
-# else(BOOST_PYTHON_SUFFIX)
-#   FIND_PACKAGE(Boost COMPONENTS python REQUIRED)
-# endif(BOOST_PYTHON_SUFFIX)
-
```
@roigcarlo I know it was not you who commented out this, but note that without this finding boost is broken for everybody. Now we are forced to define `Boost_INCLUDE_DIRS` and `Boost_LIBRARY_DIRS` instead of any of the standard FindBoost inputs.

roigcarlo replied:
Ok, fixing now

roigcarlo replied:
My suggestion would be to:

```CMake
find_package(Boost)

set(Boost_USE_STATIC_LIBS   OFF)
set(Boost_USE_MULTITHREADED ON)
set(Boost_REALPATH ON)
```

Since libs are no longer needed, pretty much any version of boost will suffice, included the system one

(that one seems to work in my computer)

jcotela replied:
Thanks! I'll try it too.

roigcarlo committed:
Enabling findboost again (for the headers)

jcotela commented:
@roigcarlo I'm also able to configure boost using your fix now :+1: 

However, I also got this warning for multiple applications (though not all??)
```
  CMake Warning (dev) at applications/trilinos_application/CMakeLists.txt:85 (target_link_libraries):
  Policy CMP0023 is not set: Plain and keyword target_link_libraries
  signatures cannot be mixed.  Run "cmake --help-policy CMP0023" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  The keyword signature for target_link_libraries has already been used with
  the target "KratosTrilinosApplication".  All uses of target_link_libraries
  with a target should be either all-keyword or all-plain.

  The uses of the keyword signature are here:

   * cmake_modules/pybind11Tools.cmake:107 (target_link_libraries)

This warning is for project developers.  Use -Wno-dev to suppress it.
```
Is this something that can be done better or do we need to switch on the policy?

roigcarlo commented:
@RiccardoRossi Commented this to me as well. Seems to be happening because the visibility policy of the shared libraries is not equal among the core and the applications. In order to fix it we must decide if we make everything PRRIVATE like in 

```CMake
target_link_libraries(Kratos PRIVATE ${KRATOS_TEST_LIBRARIES} KratosCore)
```

Or public (without the `PRIVATE`)

jcotela committed:
Updating StabilizedCFDApplication for pybind.

roigcarlo committed:
Using -DPYTHON_EXECUTABLE instead of old include and lib

roigcarlo committed:
Updating scripts

roigcarlo committed:
Sneaky improvement in travis that shoul work with pybind

jcotela committed:
Porting HDF5Application.

jcotela committed:
Incorporating recent changes in repository

jcotela commented:
I don't know if this only happens to me or if it is a known issue, but I can not run C tests with this branch. I get
```
Traceback (most recent call last):
  File "kratos_c_tests.py", line 5, in <module>
    Tester.SetVerbosity(Tester.Verbosity.TESTS_OUTPUTS)
AttributeError: type object 'Kratos.Tester' has no attribute 'Verbosity'
```

roigcarlo commented:
@jcotela Same here. It seems to be a problem in `kratos/python/add_testing_to_python.cpp`. I'll take care

jcotela committed:
Porting AdjointFluidApplication.

roigcarlo commented:
Also, I am getting:

```
Alias must point to a definition
void (%"class.Kratos::ThermalSphericParticle.350"*)* @_ZN6Kratos22ThermalSphericParticleINS_15SphericParticleEED1Ev
```

While compiling with -O3 Clang only. Gcc seems to work and other optimization levels are ok

josep-m-carbonell committed:
updating cross product

josep-m-carbonell committed:
setting properties correctly

roigcarlo committed:
Fixing error. Thx @RiccardoRossi

josep-m-carbonell committed:
add build beam string skin

josep-m-carbonell committed:
Merge branch 'master' into solid/develop

josep-m-carbonell committed:
Merge branch 'master' into solid/develop

roigcarlo committed:
Rewriting Tester interface to enable enums

roigcarlo commented:
@jcotela Last commit fixes the `verbosity` issue.
It looks like the enum syntax has greatly changed, and I detect that it was used in some other places across the interface. I believe we should recheck that all enums in the python files work as intended.

Here there is an example of how new enums that live inside a class scope need to be exported: 
https://github.com/pybind/pybind11/blob/master/tests/test_enum.cpp#L60-L64

There is one case missing which is scoped enums inside a class (which naturally is the one I wanted) which can be solved as the unscoped enum inside class without the need of `.export_values()` call.

miguelmaso committed:
porting to pybind11

miguelmaso committed:
removing other boost includes

jcotela commented:
@roigcarlo I am now able to run the tests. Thanks!

However, this has allowed me to discover more issues... One thing I could not find in the tutorial is what to do in the case of module dependencies (so far, I haven't been able to link a pybind module to another pybind module... What would be the syntax to do so?). I have been skipping them for now, since I saw that this is what was done in the trilinos application anyway, but this means that using something (typically variables) from one application from another gives me a runtime error. I can fix it easily on my application, but I'm afraid this will cause some issues, for example, in trilinos, which currently depends heavily on the fluid dynamics (and maybe others).

What is the plan for this?

jcotela committed:
HDF5/AdjointFluid: adding missing bases.

josep-m-carbonell committed:
adding first strain rate laws

josep-m-carbonell committed:
corrections to solid and pfem apps

loumalouomega committed:
Transition conflict

loumalouomega committed:
Merge branch 'master' into switching_to_std_pointer

loumalouomega committed:
Solving most of conflicts and update

loumalouomega committed:
Typo

loumalouomega committed:
Fix Object constructor

loumalouomega commented:
The ContactStructuralMechanics has been updated. Many of my test don't work because the variables look like have not being properly exported to python

loumalouomega committed:
Adding missing variable to python

loumalouomega committed:
Clean up

loumalouomega commented:
In particular the error is 

      RuntimeError: Unable to load a custom holder type from a default-holder instance

RiccardoRossi commented:
Dear @loumalouomega first of all take into account that you should not commit directly to this branch. Instead you should create a subbranch and later a PR for merging into the pybind branch. This procedure is needed due to the size of the PR.

the error you get is due to the fact that the base class defines a "custom holder" type. Since this is done any class derived from that in the python interface will have to do the same.

loumalouomega committed:
Missing typenames

philbucher commented:
@RiccardoRossi @roigcarlo from what I understand after this is merged we don't need to compile boost any more, the remaining includes (only ublas right?) are header-only ?

Will you then include it in Kratos as well until it is replaced by AMatrix? Then the user would not have to download any boost any more

loumalouomega committed:
Adding pointer to python for TableStreamUtility

josep-m-carbonell committed:
beam emc update correction

josep-m-carbonell committed:
main solid compatible with other process scripts

marandra commented:
When setting SHARED instead of MODULE in StructuralMechanicsApplication/CMakeLists.txt (L106) 
`pybind11_add_module(KratosStructuralMechanicsApplication SHARED ${KRATOS_STRUCTURAL_MECHANICS_APPLICATION_SOURCES})`

compilation throws the following error:
```
[ 53%] Built target KratosCore
[ 54%] Built target KratosExternalSolversApplication
[ 63%] Built target KratosHDF5Application
[ 77%] Built target Kratos
Scanning dependencies of target KratosStructuralMechanicsApplication
[ 78%] Building CXX object applications/StructuralMechanicsApplication/CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o
[ 78%] Linking CXX shared library KratosStructuralMechanicsApplication.cpython-35m-x86_64-linux-gnu.so
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `PyInit_KratosStructuralMechanicsApplication':
/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/custom_python/structural_mechanics_python_application.cpp:38: undefined reference to `Py_GetVersion'
/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/custom_python/structural_mechanics_python_application.cpp:38: undefined reference to `PyExc_ImportError'
/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/custom_python/structural_mechanics_python_application.cpp:38: undefined reference to `PyErr_Format'
/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/custom_python/structural_mechanics_python_application.cpp:38: undefined reference to `PyExc_ImportError'
/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/custom_python/structural_mechanics_python_application.cpp:38: undefined reference to `PyErr_SetString'
/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/custom_python/structural_mechanics_python_application.cpp:38: undefined reference to `PyExc_ImportError'
/home/mraschi/Projects/Kratos/applications/StructuralMechanicsApplication/custom_python/structural_mechanics_python_application.cpp:38: undefined reference to `PyErr_SetString'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::cast_error::set_error() const':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/detail/common.h:675: undefined reference to `PyExc_RuntimeError'
/home/mraschi/Projects/Kratos/external_libraries/pybind11/detail/common.h:675: undefined reference to `PyErr_SetString'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::error_scope::error_scope()':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/detail/common.h:714: undefined reference to `PyErr_Fetch'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::error_scope::~error_scope()':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/detail/common.h:715: undefined reference to `PyErr_Restore'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::buffer_info::~buffer_info()':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/buffer_info.h:77: undefined reference to `PyBuffer_Release'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::error_already_set::error_already_set()':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:295: undefined reference to `PyErr_Fetch'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::error_already_set::restore()':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:306: undefined reference to `PyErr_Restore'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::hasattr(pybind11::handle, char const*)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:356: undefined reference to `PyObject_HasAttrString'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::getattr(pybind11::handle, char const*)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:366: undefined reference to `PyObject_GetAttrString'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::getattr(pybind11::handle, char const*, pybind11::handle)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:381: undefined reference to `PyObject_GetAttrString'
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:384: undefined reference to `PyErr_Clear'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::setattr(pybind11::handle, pybind11::handle, pybind11::handle)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:390: undefined reference to `PyObject_SetAttr'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::setattr(pybind11::handle, char const*, pybind11::handle)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:394: undefined reference to `PyObject_SetAttrString'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::detail::accessor_policies::generic_item::get(pybind11::handle, pybind11::handle)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:497: undefined reference to `PyObject_GetItem'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::detail::accessor_policies::generic_item::set(pybind11::handle, pybind11::handle, pybind11::handle)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:503: undefined reference to `PyObject_SetItem'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::detail::accessor_policies::list_item::get(pybind11::handle, unsigned long)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:528: undefined reference to `PyList_GetItem'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::detail::accessor_policies::tuple_item::get(pybind11::handle, unsigned long)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:545: undefined reference to `PyTuple_GetItem'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::detail::accessor_policies::tuple_item::set(pybind11::handle, unsigned long, pybind11::handle)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:552: undefined reference to `PyTuple_SetItem'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::detail::iterator_policies::dict_readonly::increment()':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:663: undefined reference to `PyDict_Next'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::str::str(char const*)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:851: undefined reference to `PyUnicode_FromString'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::str::operator std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >[abi:cxx11]() const':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:868: undefined reference to `PyUnicode_AsUTF8String'
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:874: undefined reference to `PyBytes_AsStringAndSize'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::str::raw_str(_object*)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:887: undefined reference to `PyObject_Str'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::none::none()':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:967: undefined reference to `_Py_NoneStruct'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::weakref::weakref(pybind11::handle, pybind11::handle)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1060: undefined reference to `PyWeakref_NewRef'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::capsule::check_(pybind11::handle)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1084: undefined reference to `PyCapsule_Type'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::capsule::capsule(void const*, char const*, void (*)(_object*))':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1089: undefined reference to `PyCapsule_New'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::capsule::capsule(void const*, void (*)(void*))::{lambda(_object*)#1}::operator()(_object*) const':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1103: undefined reference to `PyCapsule_GetContext'
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1104: undefined reference to `PyCapsule_GetPointer'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::capsule::capsule(void const*, void (*)(void*))':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1102: undefined reference to `PyCapsule_New'
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1111: undefined reference to `PyCapsule_SetContext'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::capsule::name() const':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1132: undefined reference to `PyCapsule_GetName'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::tuple::tuple(pybind11::object const&)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1137: undefined reference to `PySequence_Tuple'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::tuple::tuple(unsigned long)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1138: undefined reference to `PyTuple_New'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::tuple::size() const':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1141: undefined reference to `PyTuple_Size'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::dict::dict()':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1150: undefined reference to `PyDict_New'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::dict::size() const':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1159: undefined reference to `PyDict_Size'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::list::list(unsigned long)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1187: undefined reference to `PyList_New'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::list::size() const':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1190: undefined reference to `PyList_Size'
CMakeFiles/KratosStructuralMechanicsApplication.dir/custom_python/structural_mechanics_python_application.cpp.o: In function `pybind11::repr(pybind11::handle)':
/home/mraschi/Projects/Kratos/external_libraries/pybind11/pytypes.h:1286: undefined reference to `PyObject_Repr'
...
```

ipouplana committed:
First changes for the port to pybind

loumalouomega committed:
Merge pull request #1744 from KratosMultiphysics/contact/correcting-pybind

[ContactStructuralMechanicsApplication][Pybind]Corrections needed

ipouplana committed:
More changes to adapt Poro for pybind

ipouplana committed:
Merge branch 'switching_to_std_pointer' into porting-poro-to-pybind

Commit:
Split Core from Application target

RiccardoRossi committed:
Merge pull request #1724 from KratosMultiphysics/porting-shallow-w-app-to-pybind

Porting shallow Water application to pybind

RiccardoRossi committed:
Merge pull request #1747 from KratosMultiphysics/uptades-to-use-application-as-base

Updates to use application as base

RiccardoRossi committed:
solving a conflict with c2c variables

roigcarlo committed:
Updating cmake

roigcarlo committed:
porting to pybind

roigcarlo committed:
Adding missing dependency

roigcarlo committed:
Enabling cotire for the structural core

roigcarlo committed:
Enabling cotire for the structural core

roigcarlo committed:
Merge pull request #1754 from KratosMultiphysics/pybind/enabling-cotire-struct-core

Enabling cotire for the structural core

ipouplana committed:
Merge branch 'switching_to_std_pointer' into porting-poro-to-pybind

ipouplana committed:
Adapting schemes...

ipouplana committed:
Porting CLs, strategies, processes and utilities to pybind

ipouplana committed:
Included model_part to avoid complation error

roigcarlo committed:
Merge pull request #1751 from KratosMultiphysics/pybind/porting-fda

Pybind/porting fda

josep-m-carbonell committed:
start new solution strategies

jcotela committed:
Defining fluid dynamics core lib and python module.

jcotela committed:
Linking StabilizedCFDApplication to KratosCore.

josep-m-carbonell committed:
reform dofs info

jcotela committed:
Adding missing declaration of custom variable type.

josep-m-carbonell committed:
print more info

ipouplana committed:
Merge branch 'switching_to_std_pointer' into porting-poro-to-pybind

jcotela committed:
Merge pull request #1757 from KratosMultiphysics/pybind11/fluid_dynamics_lib_and_module

Pybind11/fluid dynamics lib and module

ipouplana committed:
Merge branch 'switching_to_std_pointer' into porting-poro-to-pybind

josep-m-carbonell committed:
adding echo-level

jcotela committed:
Adding back test sources to the list of files to be compiled :-1:

jcotela committed:
Fixing bug in ThreadFixedSizeMemoryPool???

jcotela committed:
Compiling tests for FluidDynamicsApplication.

josep-m-carbonell committed:
solution instead of solving

josep-m-carbonell committed:
correction in time integration

adding calculate parameters to set it first time
seting parameters for schemes
correcting input and process handler json factories
correction in dynamic solid element tests

josep-m-carbonell committed:
correction in finalize solution step

ipouplana committed:
Adding dependencies of Poromechanics in the CMakeLists.txt

jcotela committed:
Renaming typedef to enable compilation with cotire.

ipouplana committed:
Minor

ipouplana committed:
Merge pull request #1756 from KratosMultiphysics/porting-poro-to-pybind

Porting Poromechanics to Pybind

ipouplana committed:
Adapting CMakeLists to pybind

RiccardoRossi committed:
substituting boost shared_ptr to std - will eventually require upstream support in amgcl

RiccardoRossi committed:
removing boost::shared_ptr dependencies

RiccardoRossi committed:
reactivating complex solver which was wrongly commented out

jcotela committed:
Merge pull request #1763 from KratosMultiphysics/pybind11/restoring-c-tests

Pybind11/restoring c tests

ipouplana committed:
Merge branch 'switching_to_std_pointer' into porting-dam-pybind

armingeiser committed:
porting shape optimization to pybind11

armingeiser committed:
porting eigen solvers application to pybind11

ipouplana committed:
Porting custom_python of DamApp...

ipouplana committed:
Porting CLs, strategies, processes and utilities of DamApp to pybind.

josep-m-carbonell committed:
updating builder and solvers

not tested compilation

josep-m-carbonell committed:
time integration container to process info

trying to manage dofs and schemes correctly
cleaning upgrade

josep-m-carbonell committed:
update scripts dofs and time integration methods

josep-m-carbonell committed:
all dofs set

josep-m-carbonell committed:
Merge branch 'solid/upgrade' into solid/pybind

josep-m-carbonell committed:
porting Solid and Constitutive

josep-m-carbonell committed:
porting Contact

josep-m-carbonell committed:
porting Pfem

josep-m-carbonell committed:
Solving Core process incompatibilities

josep-m-carbonell committed:
correction in Solid app porting

josep-m-carbonell committed:
porting Umat

josep-m-carbonell committed:
porting PfemSolid

josep-m-carbonell committed:
corrections in porting apps

note: the order in importing the modules is very important when one app depends on other. base apps must be imported first.

armingeiser committed:
add cotire command

armingeiser committed:
add cotire command

josep-m-carbonell commented:
Is there any mechanism to import base applications nedded when importing a kratos aplication in python. For example, just importing my application KratosMultiphysics is automatically imported, because is neeed and must be imported first. @KratosMultiphysics/technical-committee

dbaumgaertner committed:
removed unnecessary includes

ipouplana committed:
Adding cotire option in CMakeLists of Poromechanics and Dam

miguelmaso committed:
removing unused strategies

josep-m-carbonell committed:
solve virtual destructor core

josep-m-carbonell committed:
solve virtual destructors solid

josep-m-carbonell committed:
solve virtual destructor gid gauss points

maceligueta commented:
At least in Python 2.7 I would say it is impossible, as it constantly requests to import all those applications from the main script.

ipouplana committed:
Corrected linking error

roigcarlo committed:
Exporting missing class

roigcarlo committed:
Missing algorithm

roigcarlo committed:
Remove unused include that was causing conflicts

roigcarlo committed:
Fixing openmp loop iterator types

roigcarlo committed:
Bad signature in some bindings

roigcarlo committed:
Visibility issues in FluidDynamic

roigcarlo committed:
Visibility issues in StructuralMechanics

josep-m-carbonell committed:
include depending applications

roigcarlo committed:
PUBLIC -> PRIVATE

josep-m-carbonell committed:
corrections codacy and public to private

josep-m-carbonell committed:
required corrections

lluis-mv committed:
#include "utilities/openmp_utils.h" in add_cust_stra SolidMechanics

josep-m-carbonell committed:
Merge remote-tracking branch 'kratos/solid/pybind' into solid/pybind

josep-m-carbonell committed:
introduce opm utils include

roigcarlo committed:
Merge pull request #1779 from KratosMultiphysics/porting-eigen-solvers-application-to-pybind

porting eigen solvers application to pybind11

roigcarlo committed:
Moving cotire macro to avoid wrong definition of LINK_INTERFACE_LIBRARIES macro

roigcarlo committed:
Reverting last commit

roigcarlo committed:
Using CMP0022 policy to avoid problems with cotire + pybind

roigcarlo committed:
Using CMP0022 policy to avoid problems with cotire + pybind

josep-m-carbonell committed:
unused method

roigcarlo committed:
Merge pull request #1803 from KratosMultiphysics/pybind11/windows-bugfixs

Pybind11/windows bugfixs

roigcarlo committed:
Merge pull request #1778 from KratosMultiphysics/porting-shape-optimization-to-pybind

porting shape optimization to pybind11

roigcarlo committed:
Merge pull request #1781 from KratosMultiphysics/porting-dam-pybind

Porting Dam to pybind

roigcarlo committed:
Separated core and interface

roigcarlo committed:
Merge pull request #1816 from KratosMultiphysics/pybind11/separating-dem-core

Pybind11 - Separate DEM core and interface

roigcarlo committed:
Merge branch 'master' into switching_to_std_pointer

roigcarlo committed:
Merge branch 'master' into switching_to_std_pointer

roigcarlo committed:
Forgot to save those

roigcarlo committed:
Fixed some merge problems (that I caused my self)

roigcarlo committed:
Correcting some problems with python interfaces

roigcarlo committed:
Rewrote SimpleMortarMapperProcess interfaces

roigcarlo committed:
Remove initializer without flags

roigcarlo committed:
Reverted last change

roigcarlo committed:
Fix merge problem selecting old baseclass

roigcarlo committed:
Fixing Fastransfer process

RiccardoRossi committed:
solving a conflict

roigcarlo committed:
Accidentaly deleted file

jcotela committed:
Removing file from adjoint fluid application (merge error).

jcotela committed:
Removed interface to protected methods in HDF5::NodalSolutionStepDataIO

jcotela committed:
Updating python interface for pybind11.

jcotela committed:
fixing function using python list.

RiccardoRossi committed:
porting complete

jcotela committed:
Explicitly defining pointer holders for a few types.

RiccardoRossi committed:
Update CMakeLists.txt

adding cotire

jcotela committed:
Merge pull request #1823 from KratosMultiphysics/pybind11/thermo-mechanical

Pybind11/thermo mechanical

roigcarlo committed:
Fixing dem linking problems

roigcarlo committed:
Added exportable flags

roigcarlo committed:
Exporting prism

RiccardoRossi committed:
Merge pull request #1827 from KratosMultiphysics/porting_mixed_app

porting complete

roigcarlo committed:
Merge pull request #1830 from KratosMultiphysics/pybind11/fixiling-windows-link-problems

Pybind11/fixiling windows link problems

miguelmaso committed:
Merge pull request #1802 from KratosMultiphysics/fix-porting-shallow-water-to-pybind

Fixing porting shallow water application to Pybind

RiccardoRossi committed:
removing a method that was both static and not static

roigcarlo commented:
Dear all, I have blocked the master branch until tomorrow noon for the integration of pybind. No more code can be merged until this is in the master. This gives us time to finish some details and integrate the missing apps.

We apologize for the inconvenience

josep-m-carbonell committed:
restore core processes names over conditions

josep-m-carbonell committed:
group conditions and nodes to "entities" 

it must be generalized in the future, elements, conditions, nodes

josep-m-carbonell committed:
update branch

josep-m-carbonell committed:
Merge branch 'switching_to_std_pointer' into solid/pybind

josep-m-carbonell committed:
trailing white space

RiccardoRossi committed:
Merge pull request #1805 from KratosMultiphysics/solid/pybind

Porting Solid team applications to pybind

roigcarlo committed:
Adding missing source

roigcarlo committed:
Exposing application main class

roigcarlo committed:
Reverted change to CMakeLists and removed export api from header only file

roigcarlo committed:
Changed visivility scope of some instantiated templates

roigcarlo committed:
Solving linking problems on win for the solid

RiccardoRossi committed:
adding flag to avoid automatic linking of serialization lib

roigcarlo committed:
Fixing problem with template classes exposed in the interface

roigcarlo committed:
Merge branch 'switching_to_std_pointer' of https://github.com/KratosMultiphysics/Kratos into switching_to_std_pointer

roigcarlo reviewed (approved):
Seems to work, cross your fingers 👍
